operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java

```diff
@@ -0,0 +1,102 @@
+package org.keycloak.operator;
+
+import io.quarkus.logging.Log;
+import io.quarkus.test.junit.QuarkusTest;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.v2alpha1.KeycloakService;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+import org.keycloak.operator.v2alpha1.crds.KeycloakStatusCondition;
+import org.keycloak.operator.utils.K8sUtils;
+
+import java.time.Duration;
+import java.util.List;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.keycloak.operator.Constants.KEYCLOAK_DISCOVERY_SERVICE_PORT;
+import static org.keycloak.operator.v2alpha1.crds.KeycloakStatusCondition.READY;
+
+@QuarkusTest
+public class ClusteringE2EIT extends ClusterOperatorTest {
+
+    private KeycloakStatusCondition getCondition(List<KeycloakStatusCondition> conditions, String type) {
+        return conditions
+                .stream()
+                .filter(c -> c.getType().equals(type))
+                .findFirst()
+                .get();
+    }
+
+    @Test
+    public void testKeycloakScaleAsExpected() {
+        // given
+        var kc = K8sUtils.getDefaultKeycloakDeployment();
+        var crSelector = k8sclient
+                .resources(Keycloak.class)
+                .inNamespace(kc.getMetadata().getNamespace())
+                .withName(kc.getMetadata().getName());
+        K8sUtils.deployKeycloak(k8sclient, kc, true);
+
+        var kcPodsSelector = k8sclient.pods().inNamespace(namespace).withLabel("app", "keycloak");
+
+        Keycloak keycloak = k8sclient.resources(Keycloak.class)
+                .inNamespace(namespace)
+                .list().getItems().get(0);
+
+        // when scale it to 10
+        keycloak.getSpec().setInstances(10);
+        k8sclient.resources(Keycloak.class).inNamespace(namespace).createOrReplace(keycloak);
+
+        Awaitility.await()
+                .atMost(2, MINUTES)
+                .pollDelay(5, SECONDS)
+                .ignoreExceptions()
+                .untilAsserted(() -> {
+                    var conditions = crSelector
+                            .get()
+                            .getStatus()
+                            .getConditions();
+
+                    assertThat(getCondition(conditions, READY).getStatus()).isFalse();
```

Comment:
Please use:
https://github.com/keycloak/keycloak/blob/55315c18b146086fe2e0d1b290dd0106f5206ada/operator/src/test/java/org/keycloak/operator/utils/CRAssert.java#L28

Reply:
ok, I'll do it tomorrow 👍

Reply:
done